### PR TITLE
Fixes #11732: Implement a push / pop mechanism for audit mode

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/11732

### CODE REVIEW

File: tree/20_cfe_basics/dry_run_lib.cf [hunk 1]
```diff
@@ -82,6 +82,71 @@ bundle agent set_dry_run_mode(mode)
       "${configuration.debug} Enabling dry-run mode";
 }
 
+bundle agent dry_run_unique
```
Comment (amousset) [inline tree/20_cfe_basics/dry_run_lib.cf]: Could you add a comment to explain is is used as a datastore by push/pop_dry_run_mode?

File: tree/20_cfe_basics/dry_run_lib.cf [hunk 2]
```diff
@@ -82,6 +82,71 @@ bundle agent set_dry_run_mode(mode)
...
+{
+  vars:
+      "value" string => "1";
+      "stack" slist => {};
+}
+
```
Comment (amousset) [inline tree/20_cfe_basics/dry_run_lib.cf]: Could you add a usage doc in a comment: how to initialize the stack and use it afterwards, what happens if you use set_dry_run_mode with oush/pop, etc.

**Comment by peckpeck** [comment]
Commit modified

File: tree/20_cfe_basics/dry_run_lib.cf [hunk 3]
```diff
@@ -82,6 +82,82 @@ bundle agent set_dry_run_mode(mode)
       "${configuration.debug} Enabling dry-run mode";
 }
 
+
+# Bundle that hold global values used by push_dry_run_mode and pop_dry_run_mode.
+# Do not call directly at this would reinitialize its content.
```
Comment (amousset) [inline tree/20_cfe_basics/dry_run_lib.cf]: as

**Comment by amousset** [comment]
I think we should also modify the documentation of the dry-run lib to advise using push/pop and give an example

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**